internal/attestation/crafter/materials/csafvex.go

```diff
@@ -44,13 +45,22 @@ func NewCSAFVEXCrafter(materialSchema *schemaapi.CraftingSchema_Material, backen
 
 func (i *CSAFVEXCrafter) Craft(ctx context.Context, filepath string) (*api.Attestation_Material, error) {
 	i.logger.Debug().Str("path", filepath).Msg("decoding CSAF VEX file")
-	doc, err := csaf.Open(filepath)
-	// parse doesn't fail if the provided file is a valid JSON, but not a valid CSAF VEX file
-	if err != nil || doc.Document.Title == "" {
-		if err != nil {
-			i.logger.Debug().Err(err).Msg("error decoding file")
-		}
 
+	fh, err := os.Open(filepath)
```

Comment:
I'm probably lacking some context about why why `csaf.Open` is not a good option anymore. If I understand correctly, the purpose is to differentiate the different CSAF types, that's why here you are decoding to a `Vex` instead of the generic `CSAF` type defined in the library.
I'm not sure if this is counterproductive, since here we are overriding or duplicating some logic already present in the library.

Reply:
The issue is that `csaf.Open` returns a `csaf.CSAF` document but since we are overriding the struct with our custom type, it's not valid anymore. In any case csaf.Open only decodes the file into a csaf.CSAF document and does not perform any validation.

But, the main reason is that we working now with our own types that inherit from the library.

Reply:
This is open to debate in case we want a minimum of validation since right now VEX files only checks the `Title` is set.